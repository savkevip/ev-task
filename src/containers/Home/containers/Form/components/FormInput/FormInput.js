import React from "react";
import PropTypes from "prop-types";

const FormInput = ({ label, id, type, rows, onChange }) => (
  <div className="form-group">
    <label htmlFor={id}>{label}:</label>
    {!rows ? (
      <input
        id={id}
        type={type}
        className="form-control"
        onChange={onChange}
        placeholder={`Enter ${label}...`}
      />
    ) : (
      <textarea
        id={id}
        className="form-control"
        rows={rows}
        onChange={onChange}
        placeholder={`Enter ${label}...`}
      />
    )}
  </div>
);

FormInput.propTypes = {
  label: PropTypes.string,
  id: PropTypes.string,
  type: PropTypes.string,
  rows: PropTypes.number,
  onChange: PropTypes.func
};

export default FormInput;
