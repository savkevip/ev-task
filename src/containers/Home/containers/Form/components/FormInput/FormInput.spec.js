import React from "react";
import { shallow } from "enzyme";
import FormInput from "./FormInput";

const formData = {
  label: "Amount",
  type: "number",
  id: "amount",
  rows: 5,
  handleOnChange: jest.fn()
};

describe("FormInput", () => {
  it("should render without description", function() {
    const component = shallow(
      <FormInput
        label={formData.label}
        type={formData.type}
        id={formData.id}
        onChange={formData.handleOnChange}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("should render description", function() {
    const component = shallow(
      <FormInput
        label={formData.label}
        id={formData.id}
        rows={formData.rows}
        onChange={formData.handleOnChange}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
