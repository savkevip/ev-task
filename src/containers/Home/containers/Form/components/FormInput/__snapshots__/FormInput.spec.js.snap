// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should render description 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormInput
    id="amount"
    label="Amount"
    onChange={[Function]}
    rows={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="amount"
>
          Amount
          :
</label>,
        <textarea
          className="form-control"
          id="amount"
          onChange={[Function]}
          placeholder="Enter Amount..."
          rows={5}
/>,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Amount",
            ":",
          ],
          "htmlFor": "amount",
        },
        "ref": null,
        "rendered": Array [
          "Amount",
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "id": "amount",
          "onChange": [Function],
          "placeholder": "Enter Amount...",
          "rows": 5,
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="amount"
>
            Amount
            :
</label>,
          <textarea
            className="form-control"
            id="amount"
            onChange={[Function]}
            placeholder="Enter Amount..."
            rows={5}
/>,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Amount",
              ":",
            ],
            "htmlFor": "amount",
          },
          "ref": null,
          "rendered": Array [
            "Amount",
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "id": "amount",
            "onChange": [Function],
            "placeholder": "Enter Amount...",
            "rows": 5,
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`FormInput should render without description 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormInput
    id="amount"
    label="Amount"
    onChange={[Function]}
    type="number"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="amount"
>
          Amount
          :
</label>,
        <input
          className="form-control"
          id="amount"
          onChange={[Function]}
          placeholder="Enter Amount..."
          type="number"
/>,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Amount",
            ":",
          ],
          "htmlFor": "amount",
        },
        "ref": null,
        "rendered": Array [
          "Amount",
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "id": "amount",
          "onChange": [Function],
          "placeholder": "Enter Amount...",
          "type": "number",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="amount"
>
            Amount
            :
</label>,
          <input
            className="form-control"
            id="amount"
            onChange={[Function]}
            placeholder="Enter Amount..."
            type="number"
/>,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Amount",
              ":",
            ],
            "htmlFor": "amount",
          },
          "ref": null,
          "rendered": Array [
            "Amount",
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "id": "amount",
            "onChange": [Function],
            "placeholder": "Enter Amount...",
            "type": "number",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
